app/scripts/modules/core/src/pipeline/config/stages/managed/importDeliveryConfigStage.ts

```diff
@@ -0,0 +1,29 @@
+import { Registry } from 'core/registry';
+import { ExecutionDetailsTasks } from 'core/pipeline';
+
+import { ImportDeliveryConfigStageConfig } from './ImportDeliveryConfigStageConfig';
+import { ImportDeliveryConfigExecutionDetails } from './ImportDeliveryConfigExecutionDetails';
+import { IStageOrTriggerBeforeTypeValidationConfig } from 'core/pipeline/config/validation/stageOrTriggerBeforeType.validator';
+import { SETTINGS } from 'core/config';
+
+if (SETTINGS.feature.managedDelivery) {
+  Registry.pipeline.registerStage({
+    label: 'Import Delivery Config',
+    description:
+      "Retrieve a Delivery Config manifest from the git repository configured in the pipeline's trigger, then update it in Spinnaker.",
+    extendedDescription: `<a target="_blank" href="https://www.spinnaker.io/reference/managed-delivery/gitops.md">
+      <span class="small glyphicon glyphicon-file"></span> Documentation</a>`,
+    key: 'importDeliveryConfig',
+    restartable: false,
+    component: ImportDeliveryConfigStageConfig,
+    executionDetailsSections: [ImportDeliveryConfigExecutionDetails, ExecutionDetailsTasks],
+    //validateFn: validate,
+    validators: [
+      {
+        type: 'stageOrTriggerBeforeType',
+        stageType: 'git',
```

Comment:
just a note for the future that we'll need a way to dynamically modify this once we support Rocket triggers internally at Netflix (I don't have a great answer right now for how — we haven't really needed to worry about it in the past).

Reply:
We might want to add something like a `providesRepositoryInformation` flag to the trigger registration, similar to what we did for bakes (there is a `providesVersionForBake` flag on several triggers, and a corresponding `UpstreamVersionProvidedValidator` available).

Or we want to refactor how we handle both scenarios to make this kind of check pluggable.

Reply:
strong agree — we can look into something like that one Luis is ready.